docs/src/docs/asciidoc/guides/how-to-userinfo.adoc

```diff
@@ -0,0 +1,89 @@
+
+The preferred way to customize the user info response is to add standard claims to the `id_token`.
+The following listing shows how add claims to the `id_token`.
+
+[[sample.userinfo.idtoken]]
+include::code:IdTokenCustomizerConfig[]
+
+This configuration provides the following:
+
+<1> An instance of `OAuth2TokenCustomizer` for customizing the `id_token`.
+<2> A custom service used to obtain user info in a domain-specific way.
+
+The following listing shows a custom service for looking up user info in a domain-specific way:
+
+include::code:OidcUserInfoService[]
+
+[[customize-user-info-mapper]]
+=== Customize the User Info Mapper
+
+In order to fully customize the user info response, you can provide a custom user info mapper capable of generating the object used to render the response, which is an instance of the `OidcUserInfo` class from Spring Security.
+The mapper implementation receives an instance of `OidcUserInfoAuthenticationContext` with information about the current request, including the xref:{docs-dir}/core-model-components.adoc#oauth2-authorization-service[`OAuth2Authorization`].
+
+The following listing shows how to use the customization option that is available while working directly with the `OAuth2AuthorizationServerConfigurer`.
+
+[[sample.userinfo.jwt]]
+include::code:JwtUserInfoMapperSecurityConfig[]
+
+This configuration maps claims from the access token (which is a JWT when using the xref:{docs-dir}/getting-started.adoc#sample.gettingStarted[Getting Started config]) to populate the user info response and provides the following:
+
+<1> A Spring Security filter chain for the xref:{docs-dir}/protocol-endpoints.adoc[Protocol Endpoints].
+<2> A user info mapper that maps claims in a domain-specific way.
+<3> An example showing the configuration option for customizing the user info mapper.
+<4> Resource server support that allows User Info requests to be authenticated with access tokens.
+<5> An example showing how to apply the `OAuth2AuthorizationServerConfigurer` to the Spring Security configuration.
+
+The user info mapper is not limited to mapping claims from a JWT, but this is a simple example that demonstrates the customization option.
+Similar to the <<customize-id-token,example above>> where we customize claims of the ID token, you can customize claims of the access token itself ahead of time, as in the following example:
+
+include::code:JwtTokenCustomizerConfig[]
+
+Whether you customize the user info response directly or use this example and customize the access token, you can look up information in a database, perform an LDAP query, make a request to another service, or any other means of obtaining the information you want to be presented in the user info response.
```

Comment:
Change `or any` to `or use any`

(That makes each item in the list start with a verb, for consistency.)